File: frontend/src/app/shared/models/address.ts
export interface AddressResponse {
  addressId: number;
  userId: number | null;
  address: string | null;
  city: string | null;
  state: string | null;
  pincode: string | null;
}

export interface CreateAddressRequest {
  userId?: number;
  address: string;
  city: string;
  state: string;
  pincode: string;
}

export interface UpdateAddressRequest {
  userId?: number;
  address: string;
  city: string;
  state: string;
  pincode: string;
}

File: backend/Models/Address.cs
using System;
using System.Collections.Generic;

namespace backend.Models;

public partial class Address
{
    public int AddressId { get; set; }

    public int? UserId { get; set; }

    public string? Address1 { get; set; }

    public string? City { get; set; }

    public string? State { get; set; }

    public string? Pincode { get; set; }

    public virtual User? User { get; set; }
}
